### PR TITLE
DO NOT MERGE: fixing verification bug

### PR DESCRIPTION
**Pull request by rhiGriff**
This PR makes 2 small changes:

The first change removes the setting of a users account to verified in the django server when they change a password.

The second stops a change password email being sent to a user if their account is not in the ACTIVE state in the party database.

### CODE REVIEW

File: ras_party/controllers/account_controller.py [hunk 1]
```diff
@@ -275,23 +274,24 @@ def request_password_change(payload, session):
...
-        'RESET_PASSWORD_URL': verification_url,
-        'FIRST_NAME': respondent.first_name
-    }
+        personalisation = {
+            'RESET_PASSWORD_URL': verification_url,
+            'FIRST_NAME': respondent.first_name
+        }
 
-    logger.info('Reset password url', url=verification_url)
+        logger.info('Reset password url', url=verification_url)
```
Comment (jcox-dev) [inline ras_party/controllers/account_controller.py]: Maybe we can log the party_id with the generated verification_url here?

**Review by jcox-dev — CHANGES_REQUESTED** [review]
Does and checks what it says on the tin. 

Included a suggestion for linking the party with the notify URL in the logs.

**Comment by rhiGriff** [comment]
[Receiving 500 error when trying to verify email address.](https://trello.com/c/Zpr07DyQ/221-receiving-500-error-when-trying-to-verify-email-address)

File: ras_party/controllers/account_controller.py [hunk 2]
```diff
@@ -275,23 +274,24 @@ def request_password_change(payload, session):
     if not respondent:
         raise RasError("Respondent does not exist.", status=404)
 
-    email_address = respondent.email_address
-    verification_url = PublicWebsite().reset_password_url(email_address)
+    if respondent.status is RespondentStatus.ACTIVE:
```
Comment (gemmai95) [inline ras_party/controllers/account_controller.py]: I think we need some debug level logging on this. Just a "Requesting password change" and "Successfully changed password" with party_id potentially

File: test/test_respondent_controller.py [hunk 3]
```diff
@@ -55,6 +55,24 @@ def populate_with_respondent(self, tran, session, respondent=None):
         account_controller.register_user(respondent, tran)
         return self.respondent
 
+    @transactional
+    @with_db_session
+    def populate_with_active_respondent(self, tran, session, respondent=None):
```
Comment (gemmai95) [inline test/test_respondent_controller.py]: I don't know that a completely new function to add an active respondent is necessary just pass it in to the populate with respondent that's why it has the respondent.get('status'). This seems unnecessary when the same can be done with code already there.

File: ras_party/controllers/account_controller.py [hunk 4]
```diff
@@ -275,23 +274,29 @@ def request_password_change(payload, session):
...
+        try:
+            NotifyGateway(current_app.config).request_password_change(
+                email_address, personalisation, str(party_id))
+        except RasNotifyError:
+            # Note: intentionally suppresses exception
+            logger.error('Error sending notification email for party_id', party_id=party_id)
+
+        logger.debug('Password reset email successfully sent', party_id=respondent.party_uuid)
 
     return {'response': "Ok"}
```
Comment (tej99) [inline ras_party/controllers/account_controller.py:301]: Should we still be returning a response ok when we haven't actually done what the call is expecting us to do (send a password change link)?

File: test/test_respondent_controller.py (hunk 3 above)
Comment (jcox-dev) [reply test/test_respondent_controller.py]: 👍 Could be a partial instead (if we updated the params for `populate_with_respondent`:
```python
...
    populate_with_active_respondent = functools.partialmethod(populate_with_respondent, status=RespondentStatus.ACTIVE)
...
```

File: ras_party/controllers/account_controller.py (hunk 4 above)
Comment (rhiGriff) [reply ras_party/controllers/account_controller.py:301]: Yes this is as Dan has said it should carry on to the same screen etc just not send the email

File: test/test_respondent_controller.py (hunk 3 above)
Comment (gemmai95) [reply test/test_respondent_controller.py]: We have an active respondent but yeah we could.

**Review by jcox-dev — APPROVED** [review]

**Review by gemmai95 — APPROVED** [review]

**Review by tej99 — APPROVED** [review]

**Review by liamtoozer — APPROVED** [review]

**Pull request merged**